CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+  Use `&&=` to preprocess variables that may or may not exist. Using `&&=`
+  will change the value only if it exists, removing the need to check its
+  existence with `if`.
+<sup>[[link](#double-amper-preprocess)]</sup>
+
+  ```Ruby
+  # bad
+  if something
+    something = something.downcase
+  end
+
+  # bad
+  something = something ? something.downcase : nil
+
+  # ok
+  something = something.downcase if something
+
+  # good
+  something = something && something.downcase
+
+  # better
+  something &&= something.downcase
+  ```
+
+* <a name="no-case-equality"></a>
+  Avoid explicit use of the case equality operator `===`. As its name implies
+  it is meant to be used implicitly by `case` expressions and outside of them it
+  yields some pretty confusing code.
+<sup>[[link](#no-case-equality)]</sup>
+
+  ```Ruby
+  # bad
+  Array === something
+  (1..100) === 7
+  /something/ === some_string
+
+  # good
+  something.is_a?(Array)
+  (1..100).include?(7)
+  some_string =~ /something/
+  ```
+
+* <a name="eql"></a>
+  Do not use `eql?` when using `==` will do. The stricter comparison semantics
+  provided by `eql?` are rarely needed in practice.
+<sup>[[link](#eql)]</sup>
+
+  ```Ruby
+  # bad - eql? is the same as == for strings
+  "ruby".eql? some_str
+
+  # good
+  "ruby" == some_str
+  1.0.eql? x # eql? makes sense here if want to differentiate between Fixnum and Float 1
+  ```
+
+* <a name="no-cryptic-perlisms"></a>
+  Avoid using Perl-style special variables (like `$:`, `$;`, etc. ). They are
+  quite cryptic and their use in anything but one-liner scripts is discouraged.
+  Use the human-friendly aliases provided by the `English` library.
+<sup>[[link](#no-cryptic-perlisms)]</sup>
+
+  ```Ruby
+  # bad
+  $:.unshift File.dirname(__FILE__)
+
+  # good
+  require 'English'
+  $LOAD_PATH.unshift File.dirname(__FILE__)
+  ```
+
+* <a name="parens-no-spaces"></a>
+  Do not put a space between a method name and the opening parenthesis.
+<sup>[[link](#parens-no-spaces)]</sup>
+
+  ```Ruby
+  # bad
+  f (3 + 2) + 1
+
+  # good
+  f(3 + 2) + 1
+  ```
+
+* <a name="parens-as-args"></a>
+  If the first argument to a method begins with an open parenthesis, always
+  use parentheses in the method invocation. For example, write `f((3 + 2) + 1)`.
+<sup>[[link](#parens-as-args)]</sup>
+
+
+* <a name="no-nested-methods"></a>
+  Do not use nested method definitions, use lambda instead.
+  Nested method definitions actually produce methods in the same scope
+  (e.g. class) as the outer method. Furthermore, the "nested method" will be
+  redefined every time the method containing its definition is invoked.
+<sup>[[link](#no-nested-methods)]</sup>
+
+  ```Ruby
+  # bad
+  def foo(x)
+    def bar(y)
+      # body omitted
+    end
+
+    bar(x)
+  end
+
+  # good - the same as the previous, but no bar redefinition on every foo call
+  def bar(y)
+    # body omitted
+  end
+
+  def foo(x)
+    bar(x)
+  end
+
+  # also good
+  def foo(x)
+    bar = ->(y) { ... }
```

Comment:
I think the only big place we use lambda is in the date filter, and that I wrote it with lambdas gives me great regrets (it's hard to read, hard to maintain).